### PR TITLE
@ #120 | should add dev-setup

### PR DESCRIPTION
**Pull request by hieptranquoc**
connect #120

### CODE REVIEW

**Comment by hieptranquoc** [comment]
@hoatle review this PR for me!

File: dev-setup/chef/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+version: '3'
+
+services:
+
+  dev:
+    image: chef/chefdk
```
Comment (hoatle) [inline dev-setup/chef/docker-compose.yml]: use teracy/chefdk image

File: dev-setup/chef/main-cookbooks/blog-cookbook/.delivery/project.toml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Delivery Prototype for Local Phases Execution
```
Comment (hoatle) [inline dev-setup/chef/main-cookbooks/blog-cookbook/.delivery/project.toml]: use "teracy-blog" as cookbook name

File: dev-setup/vagrant_config_default.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      "workspace/blog/dev-setup/chef/main-cookbooks"
+    ],
+    "_ua_run_list": [
+      "blog-cookbook"
+    ],
+    "json": {
+      "teracy-dev": {
+        "env_vars": [{
+          "_id": "blog-cookbook-0",
+          "key": "KITCHEN_LOCAL_YAML",
```
Comment (hoatle) [inline dev-setup/vagrant_config_default.json]: no need

**Comment by hieptranquoc** [comment]
@hoatle fixed! review this PR again 💃

**Comment by hoatle** [comment]
@hieptranquoc this is not enough, make sure that:

```
$ cd ~/teracy-dev/workspace
$ git clone <forked_repo>
$ vagrant hostmanager
$ vagrant reload --provision
```

and then we should open dev.blog.teracy.dev to view the blog site on local dev mode.

make use we will have local prod and local review modes, too. (blog.teracy.dev, review.blog.teracy.dev)

You need to look into many existing projects to understand the workflow and apply for this project, just look and learn and adapt.

**Comment by hoatle** [comment]
My previous comment should be addressed, too:

```
still not good enough, need to check out the similar projects to:

update README
have properly configuration
have upgrade guide
```

**Comment by hoatle** [comment]
@hieptranquoc review done, please check and do more carefully, seems that you're out of focus.

**Comment by hieptranquoc** [comment]
@hoatle i fixed! please, review this PR again! 💃

File: docker-compose.prod.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+version: "3"
+
+services:
+
+  prod:
+    container_name: teracy-blog_prod
```
Comment (hoatle) [inline docker-compose.prod.yml]: remove container_name (because docker-compose scale will not work with this)

File: docker-compose.prod.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    build:
+      context: .
+      dockerfile: Dockerfile
+      args:
+        CI_BUILD_ID: ${CI_BUILD_ID}
```
Comment (hoatle) [inline docker-compose.prod.yml]: use less build args:

ARG CI_BUILD_ID
ARG CI_BUILD_REF
ARG CI_REGISTRY_IMAGE
ARG CI_BUILD_TIME

are enough

File: docker-compose.review.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
+version: "3"
+
+services:
+
+  review:
+    container_name: teracy-blog_review
```
Comment (hoatle) [inline docker-compose.review.yml]: remove container_name

File: README.markdown [hunk 7]
```diff
@@ -13,34 +13,38 @@ Teracy's official blog at http://blog.teracy.com
...
-  $ cd teracy-blog
+  $ git clone <your_forked_repository_here>
+  $ cd blog
   $ git remote add upstream git@github.com:teracyhq/blog.git
   ```
 Note: You need to fetch the latest changes of `teracy-blog` before going to the next step. Please see the details at http://dev.teracy.org/docs/develop/workflow.html.
 
 ## How to work in dev mode
 
-- Make sure your `vagrant_config_override.json` has the following config:
```
Comment (hoatle) [inline README.markdown]: remove this section because users are not required to know about this on this README file

**Comment by hoatle** [comment]
@hieptranquoc view done, please make sure:
- check my comments
- have upgrade docs
- follow this best practices: https://github.com/teracyhq/dev/issues/395

File: README.markdown (hunk 7 above)
Comment (hieptranquoc) [reply README.markdown]: fixed!

File: docker-compose.review.yml (hunk 6 above)
Comment (hieptranquoc) [reply docker-compose.review.yml]: removed!

File: docker-compose.prod.yml (hunk 5 above)
Comment (hieptranquoc) [reply docker-compose.prod.yml]: fixed!

File: docker-compose.prod.yml (hunk 4 above)
Comment (hieptranquoc) [reply docker-compose.prod.yml]: removed!

**Comment by hieptranquoc** [comment]
@hoatle i updated ! please check PR again, please!

File: README.markdown [hunk 8]
```diff
@@ -13,63 +13,25 @@ Teracy's official blog at http://blog.teracy.com
 
   ```bash
   $ cd ~/teracy-dev/workspace/
-  $ git clone <your_forked_repository_here> teracy-blog
-  $ cd teracy-blog
+  $ git clone <your_forked_repository_here>
```
Comment (hoatle) [inline README.markdown]: git clone <your_forked_repository_here> blog

File: README.markdown [hunk 9]
```diff
@@ -13,63 +13,25 @@ Teracy's official blog at http://blog.teracy.com
...
-```
-
-- `$ vagrant reload --provision` and `$ vagrant hostmanager` to make sure everything is updated.
-
-- Start:
-
-  Keep the first terminal window and let the file watching keep running.
+- Dev mode was installed by default in `teracy-blog` cookbook.
 
+- `$ vagrant hostmanager` to update host file.
```
Comment (hoatle) [inline README.markdown]: update the hosts file

**Comment by hoatle** [comment]
ready for @hoavt to review and make sure all the requirements must be met.

File: README.markdown (hunk 8 above)
Comment (hieptranquoc) [reply README.markdown]: fixed!

File: README.markdown (hunk 9 above)
Comment (hieptranquoc) [reply README.markdown]: updated!

File: README.markdown [hunk 10]
```diff
@@ -13,63 +13,25 @@ Teracy's official blog at http://blog.teracy.com
...
-  }]
-}
-```
-
-- `$ vagrant reload --provision` and `$ vagrant hostmanager` to make sure everything is updated.
-
-- Start:
-
-  Keep the first terminal window and let the file watching keep running.
+- Dev mode was installed by default in `teracy-blog` cookbook.
```
[the remark this answers is not in the document]
Comment (hoavt) [reply README.markdown]: To preview the blog, do I need to care for the `teracy-blog` cookbook?

File: README.markdown [hunk 11]
```diff
@@ -13,63 +13,25 @@ Teracy's official blog at http://blog.teracy.com
...
-  ```bash
-  $ cd ~/teracy-dev
-  $ vagrant ssh
-  $ ws
-  $ cd teracy-blog
-  $ docker-compose pull && docker-compose up -d dev
-  ```
-
-  Open dev.blog.teracy.dev to preview on local.
+  Open http://dev.blog.teracy.dev or https://dev.blog.teracy.dev to preview on local.
```
Comment (hoavt) [inline README.markdown]: I follow steps above, but when opening these URLs, This site can’t be reached

File: README.markdown (hunk 10 above)
[the remark this answers is not in the document]
Comment (hieptranquoc) [reply README.markdown]: @hoavt no, you do not need to care about `teracy-blog` cookbook

File: dev-setup/chef/main-cookbooks/teracy-blog/README.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
+# teracy-blog dev-setup
+
+You can use other software provisioning tools other than Chef, see: https://www.vagrantup.com/docs/provisioning/
+`teracy-dev` supports all the configuration of vagrant supported provisioners into the
+`vagrant_config_override.json` file, you can combine them for the dev-setup.
+
+
+## dev-setup/chef/main-cookbooks/teracy-blog cookbook
+
+- The teracy-blog Chef project was initial generated by the following commands:
```
[the remark this answers is not in the document]
Comment (hieptranquoc) [reply dev-setup/chef/main-cookbooks/teracy-blog/README.md]: i fixed! but you do not need to care about teracy-blog cookbook

File: README.markdown [hunk 13]
```diff
@@ -13,63 +13,25 @@ Teracy's official blog at http://blog.teracy.com
...
 
   Need to make sure the blog build completed before previewing it. Use the command below:
 
   ```bash
+  $ ws
```
[the remark this answers is not in the document]
Comment (hieptranquoc) [reply README.markdown]: @hoavt i fixed!

File: README.markdown (hunk 11 above)
Comment (hieptranquoc) [reply README.markdown]: @hoavt i fixed!

File: README.markdown [hunk 14]
```diff
@@ -13,63 +13,27 @@ Teracy's official blog at http://blog.teracy.com
```
Comment (hoavt) [inline README.markdown]: Should change "Set up teracy-dev if not yet, follow: http://dev.teracy.org/docs/develop/getting_started.html" --> 
"Set up teracy-dev and dev-setup if not yet. Follow the document at https://github.com/teracyhq/dev-setup/blob/develop/README.md."

**Comment by hieptranquoc** [comment]
@hoatle please check this PR again, please!

**Comment by hoatle** [comment]
assigned to @hoavt for reviewing.

**Comment by hoavt** [comment]
@hieptranquoc: Need to refer to the link https://github.com/teracyhq/blog/pull/123/files#diff-2e97ce00988fa50a9cd1783ea44375e7 in the doc blog/README with the workflow as https://github.com/teracyhq/dev/issues/395#issue-251179575

File: README.markdown [hunk 15]
```diff
@@ -5,71 +5,35 @@ Teracy's official blog at http://blog.teracy.com
...
 
   ```bash
   $ cd ~/teracy-dev/workspace/
-  $ git clone <your_forked_repository_here> teracy-blog
-  $ cd teracy-blog
+  $ git clone <your_forked_repository_here> blog
+  $ cd blog
   $ git remote add upstream git@github.com:teracyhq/blog.git
   ```
 Note: You need to fetch the latest changes of `teracy-blog` before going to the next step. Please see the details at http://dev.teracy.org/docs/develop/workflow.html.
```
Comment (hoavt) [inline README.markdown]: Should add this tep below Note section:
- Install the teracy-blog dev-setup once only. Follow the document at https://github.com/teracyhq/blog/dev-setup/chef/main-cookbooks/teracy-blog/README.md

(Need to make sure this link will work after the pr is merged)

Comment (hoatle) [reply README.markdown]: https://github.com/teracyhq/blog/dev-setup/README.md is the right file, not https://github.com/teracyhq/blog/dev-setup/chef/main-cookbooks/teracy-blog/README.md

Comment (hieptranquoc) [reply README.markdown]: fixed!

**Comment by hieptranquoc** [comment]
@hoavt please help me check this PR again!

File: dev-setup/README.md [hunk 16]
```diff
@@ -0,0 +1,189 @@
+# teracy-blog
+
+Teracy's official blog at http://blog.teracy.com
+
+
+## Getting Started
+
+- Set up teracy-dev and dev-setup if not yet. Follow the document at https://github.com/teracyhq/dev-setup/blob/develop/README.md.
+
+- Fork this repo into your github account
```
Comment (hoavt) [inline dev-setup/README.md]: Should have a Full stop at the end of the sentence --> "github account."

File: dev-setup/README.md [hunk 17]
```diff
@@ -0,0 +1,189 @@
...
+  ```bash
+  $ cd ~/teracy-dev/workspace/
+  $ git clone <your_forked_repository_here> blog
+  $ cd blog
+  $ git remote add upstream git@github.com:teracyhq/blog.git
+  ```
+  
+- Fetch the latest changes of `teracy-blog` before going to the next step. Please see the details at http://dev.teracy.org/docs/develop/workflow.html
+
+- Add configuration path to `vagrant_config_default.json` in `teracy/workspace/dev-setup`, if is not exist. It should look like similar this:
```
Comment (hoavt) [inline dev-setup/README.md]: Add the configuration path below to the `teracy-dev/workspace/dev-setup/vagrant_config_default.json` file, if it does not exist yet. It should look like similar to this:

Comment (hieptranquoc) [reply dev-setup/README.md]: updated!

File: dev-setup/README.md (hunk 16 above)
Comment (hieptranquoc) [reply dev-setup/README.md]: FIXED!

File: README.markdown [hunk 18]
```diff
@@ -1,177 +1,3 @@
...
-
-  ```bash
-  $ docker-compose exec dev /bin/bash
-  app@a4e5c4766cd3:/opt/app$
-  ```
-
-## Contributing
-
-Please see the CONTRIBUTING.md file.
+https://github.com/teracyhq/blog/tree/master/dev-setup/README.md
```
[the remark this answers is not in the document]
Comment (hieptranquoc) [reply README.markdown]: updated!

**Pull request merged**